anna/my-app/src/components/infiniteScroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import IMovie from '../../components/movieList/movie-card/IMovie';
+import * as QueryString from "query-string";
+import { ISearchParams } from '../../Api';
+import { LoadMoreMoviesRequestedAction } from '../../store/redux/moviesActions';
+
+interface IInfScrollState {
+  isLoadingMoreData: boolean
+}
+
+interface InfScrollProps {
+  movies: IMovie[],
+  loadMoreMovies: (searchParams: ISearchParams) => LoadMoreMoviesRequestedAction,
+  search: string
```

Comment:
Your component is too specific. Try to do components more reusable and abstract. InfinityScroll shouldn't know about the data you are loading. Now it is a movie, but you can load smth. else, collections, songs other data. 

So, 
1) Don't pass the whole array of movies, you don't use it. You use only length of it. So pass variable like currentCount or smth like that. 
2) better to name **onLoadMore** instead of loadMoreMovies.
3) I believe you don't need to pass search. You should pass function that will be called onLoad and you can use seacrh inside of it. 
4) You need **total** prop, how will you know that no movies left and you don't need to load more ? You have to know when to stop.